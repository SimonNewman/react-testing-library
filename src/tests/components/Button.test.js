import "@testing-library/jest-dom";

import { fireEvent, render, screen } from "@testing-library/react";

import Button from "@/components/Button";
import renderer from "react-test-renderer";

const buttonText = "My Test Button";

test("Renders Button component and simulates click", async () => {
  // Mock function for button onClick
  const handleClick = jest.fn(() => console.log("Button was clicked"));

  // Render the Button component
  render(<Button onClick={handleClick}>{buttonText}</Button>);

  // Get the button element
  const button = screen.getByText(buttonText);

  // Simulate a user clicking the button
  fireEvent.click(button);

  // Verify click handler was called
  expect(handleClick).toHaveBeenCalled();

  // Snapshot
  const tree = renderer.create(<Button>{buttonText}</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
