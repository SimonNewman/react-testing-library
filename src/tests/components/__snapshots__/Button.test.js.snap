// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Button component and simulates click 1`] = `
<button>
  My Test Button
</button>
`;
